libsupport/include/katana/EntityTypeManager.h

```diff
@@ -78,6 +83,181 @@ class EntityTypeManager {
         atomic_entity_type_id_to_entity_type_ids_(
             std::move(atomic_entity_type_id_to_entity_type_ids)) {}
 
+  template <typename ArrowType>
+  struct PropertyColumn {
+    int field_index;
+    std::shared_ptr<ArrowType> array;
+
+    PropertyColumn(int i, std::shared_ptr<ArrowType>& a)
+        : field_index(i), array(a) {}
+  };
+
+  /// This function can be used to convert "old style" graphs (storage format 1,
+  /// where types are represented by bool or uint8 properties) and "new style"
+  /// graphs (version > 2, where types are represented in our native type
+  /// represenation). This function is serial but it likely iterates over
+  /// O(nodes) and O(edges) vectors, so it is very slow. It should only be used
+  /// for updating old graphs.
+  ///
+  /// The number of entries in entity_type_ids should be equal to topo_size and
+  /// properties->num_rows().
+  template <template <typename> class Vector>
+  static Result<void> AssignEntityTypeIDsFromProperties(
```

Comment:
Might wanna move this big ass function into a cpp. In fact, in a later PR, I might break it up further just because I don't like long methods.

Reply:
The issue with putting it in a cpp is that it is templated and the type that we really care about passing as the template parameter is in libgalois. I will see what I can do to break out some helpers and put them in a cpp.

Reply:
yes, I did realize it later on that it's a template function.